src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentValidatorUTest.java

```diff
@@ -86,46 +86,46 @@ public void setup() {
 
     @Test
     public void validateAttendanceAssignmentHappyPath() throws IOException, AttendanceAssignmentException {
-        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID+"")).thenReturn(assignmentOptional);
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, Long.toString(CANVAS_ASSIGNMENT_ID))).thenReturn(assignmentOptional);
         assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
         Assert.assertNotEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
         Assert.assertNotEquals(AttendanceAssignment.Status.CANVAS_AND_DB_DISCREPANCY, attendanceAssignment.getStatus());
     }
 
     @Test
     public void validateCanvasAssignmentHappyPath() throws IOException, AttendanceAssignmentException {
-        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID+"")).thenReturn(assignmentOptional);
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, Long.toString(CANVAS_ASSIGNMENT_ID))).thenReturn(assignmentOptional);
 
         assignmentValidator.validateCanvasAssignment(assignmentConfigurationFromSetup, COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
         Assert.assertNotEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
         Assert.assertNotEquals(AttendanceAssignment.Status.CANVAS_AND_DB_DISCREPANCY, attendanceAssignment.getStatus());
     }
 
     @Test
-    public void validateAttendanceAssignmentEmptyOptionalIdValidationError() throws IOException, AttendanceAssignmentException {
+    public void validateAttendanceAssignmentEmptyOptionalIdValidationError() throws Exception {
         assignmentOptional = Optional.empty();
-        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID + "")).thenReturn(assignmentOptional);
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, Long.toString(CANVAS_ASSIGNMENT_ID))).thenReturn(assignmentOptional);
 
         assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
         Assert.assertEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
     }
 
 
     @Test
-    public void validateAttendanceAssignmentNullCanvasIDError() throws IOException, AttendanceAssignmentException {
+    public void validateAttendanceAssignmentNullCanvasIDError() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00112)